.github/workflows/staging.yml

```diff
@@ -0,0 +1,48 @@
+name: Staging
```

Comment:
Given that this isn't really a staging workflow, maybe it would be better to just call in "Linting" or something similar.

Reply:
Good idea, thanks. I'll also change the file name. Generally, are this and the `hugo.yml` written correctly?

Reply:
I think so. I haven't used the npm caching, but that part looks plausibly correct. I'd prefer if both of them ran on all pushes, as I don't usually make PRs until I think the branch is ready, but that isn't a strong opinion.